lib/rules/no-error-cause.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @author Sosuke Suzuki <https://github.com/sosukesuzuki>
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict"
+
+const { CONSTRUCT, READ, ReferenceTracker } = require("eslint-utils")
+
+/**
+ * @typedef {import("estree")} Node
+ * @typedef {import("estree").ClassExpression | import("estree").ClassDeclaration} ClassNode
+ * @typedef {import("estree").CallExpression} CallExpression
+ */
+
+const traceMap = {}
+for (const errorConstructorName of ["Error", "AggregateError"]) {
```

Comment:
Should we also target EvalError, RangeError, ReferenceError, SyntaxError, TypeError, and URIError?
(I may have misunderstood the specs.)

https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-native-error-types-used-in-this-standard
https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-nativeerror